src/daemon.rs

```diff
@@ -30,18 +68,236 @@ use crate::{deployment::create_owned_deployment, error::ReconcileError};
 //         .trace_id()
 // }
 
-struct Context {
-    client: Client,
+// this is actually used to pass clients around
+#[allow(dead_code)]
+pub struct Context {
+    pub client: Client,
+}
+
+#[instrument(skip(ctx))]
+async fn find_exit_node_from_label(ctx: Arc<Context>, query: &str) -> Option<ExitNode> {
+    let nodes: Api<ExitNode> = Api::all(ctx.client.clone());
+    let node_list = nodes.list(&ListParams::default().timeout(30)).await.ok()?;
+    node_list.items.into_iter().find(|node| {
+        node.metadata
+            .name
+            .as_ref()
+            .map(|name| name == query)
+            .unwrap_or(false)
+    })
+}
+#[instrument(skip(ctx))]
+async fn find_exit_node_provisioner_from_label(
```

Comment:
so if two provisioners have the same name in two separate namespaces, GFL

this is poorly thought through, either the provisioner needs to be non-namespaced, the annotation should be to a namespace + name, or we should just have the provisioner reference be in the ExitNode CRD (instead of just a label)